gobblin-runtime/src/test/java/gobblin/runtime/locks/JobLockTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime.locks;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.Level;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.runtime.locks.FileBasedJobLock;
+import gobblin.runtime.locks.JobLockEventListener;
+
+
+/**
+ * Base for all {@link JobLock} unit tests.
+ *
+ * @author Joel Baranick
+ */
+@Test(groups = {"gobblin.runtime"})
+public abstract class JobLockTest {
+
+    public abstract JobLock getJobLock() throws JobLockException;
+
+    public void testLocalJobLock() throws Exception {
```

Comment:
can you change the name of this method? shouldn't this method be tagged with the `Test` annotation?